tests-legacy/Unit/Core/Localization/Specification/FactoryTest.php

```diff
@@ -214,6 +214,90 @@ public function testBuildPriceSpecificationWithMax($data, $expected)
         $this->assertEquals($specification->getMaxFractionDigits(), $maxFractionDigits);
     }
 
+    /**
+     * Given a valid CLDR locale
+     * Given a Max Fractions digits to display in a number's decimal
+     * Given a boolean to define if we should group digits in a number's integer part
+     * Given an integer to specify max fraction digits
+     * Then calling buildPriceSpecification() should return an NumberSpecification
+     *
+     * @dataProvider getPriceData
+     */
+    public function testBuildPriceSpecificationWithPrecisionFallback(array $data, array $expected): void
+    {
+        // if maxFractionDigits < minFractionDigits, minFractionDigits = maxFractionDigits
+        // see PrestaShop\PrestaShop\Core\Localization\Specification\Number
+        // The dataProvider provides minFractionDigits === 2
+        $precisions = [
```

Comment:
Why not using a dataProvider instead https://phpunit.readthedocs.io/en/9.5/writing-tests-for-phpunit.html?highlight=dataprovider#data-providers

Reply:
@samberrry Could you do it ? We can send your PR after this.

Reply:
@Progi1984 let me check that.

Reply:
done